### PR TITLE
Add and configure Trivy for security scans

### PR DESCRIPTION
**Pull request by n1koo**
Add [trivy](https://trivy.dev) to be able to:

- Check packages for known vulnerabilities
- Check for any potential misconfigurations in configs
- Check that licenses are properly matching

Runs both in CI and CD - idea being that CD runs it agains the build images and CI just for the repo.

Theres a `.trivyignore.yaml` for tuning findings. Currently couple HIGH and one CRITICAL but none of them seem too applicable to our use case.

Example from [build pipeline](https://github.com/opencrvs/opencrvs-core/actions/runs/7800090618/job/21272357025)

Also:

- Use `node` user where we can (to make `trivy` happy about it but its also a nice security practice)
- Pull latest OS packages with `apt-get upgrade` or equivalent during builds

### CODE REVIEW

**Review by rikukissa — APPROVED** [review]

**Comment by euanmillar** [comment]
@n1koo we probably also need a PR to address the Farajaland Dockerfile? https://github.com/opencrvs/opencrvs-farajaland/blob/develop/Dockerfile and maybe also Github workflows in that repository?

**Comment by n1koo** [comment]
@euanmillar @rikukissa have another look at the change - specifically `trivy.yaml` and `.trivyignore.yam`.

Nothing major in those but those exclusion rules etc is what we will eventually clear out by doing upgrades. Nothing house-on-fire urgent

File: trivy.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+severity:
+  - HIGH
+  - CRITICAL
+scan:
+  skip-dirs:
+    - node_modules
+    - .secrets
+    - data
+    - development-environment
+    - sequece-diagrams
```
Comment (naftis) [inline trivy.yaml:19]:
```suggestion
    - sequence-diagrams
```

File: trivy.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# & Healthcare Disclaimer located at http://opencrvs.org/license.
+#
+# Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+exit-code: 1
+severity:
+  - HIGH
+  - CRITICAL
+scan:
+  skip-dirs:
+    - node_modules
```
Comment (naftis) [inline trivy.yaml:15]: curious, does Trivy build core, so are node_modules, data, and the yarn folder visible to it?

Comment (n1koo) [reply trivy.yaml:15]: You are right, it is for local running but also for the image scanning mode where build has happened :)

**Pull request merged**